include/input_state.h

```diff
@@ -27,34 +28,55 @@ enum pointer_values {
 	M_SCROLL_DOWN = 276,
 };
 
+enum pointer_mode {
+	// Target
+	M_FLOATING = 1 << 0,
+	M_TILING = 1 << 1,
+	// Action
+	M_DRAGGING = 1 << 2,
+	M_RESIZING = 1 << 3,
```

Comment:
Style: I prefer to declare these with powers of two explicitly, like so:

    M_FLOATING = 1,
    M_TILING = 2,
    M_DRAGGING = 4,
    M_RESIZING = 8